src/libraries/Native/Unix/System.Globalization.Native/pal_localeStringData.c

```diff
@@ -221,28 +224,59 @@ int32_t GlobalizationNative_GetLocaleInfoString(const UChar* localeName,
     switch (localeStringData)
     {
         case LocaleString_LocalizedDisplayName:
-            uloc_getDisplayName(locale, DetectDefaultLocaleName(), value, valueLength, &status);
+            assert(uiLocaleName != NULL);
```

Comment:
The `uiLocaleName` was defaulted to `null` in all the calls to the P/Invoke. Since we are asserting it is not `NULL` here, would it be better to not have the `= null` on the C# side and force callers to explicitly pass `null` in?

Reply:
The pinvoke is not the issue but the wrapper of the pinvoke will need to do the same (make the parameter not defaulted). If we do the wrapper too, that means we are going to touch too many places in the code, almost all callers of `IcuGetLocaleInfo` and `GetLocaleInfoCore`. I don't think we need to go to that level while we are asserting on the concerned cases.

Reply:
OK, that makes sense to me.